build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/extractor/DockerPush.java

```diff
@@ -0,0 +1,114 @@
+import org.jfrog.build.extractor.clientConfiguration.ArtifactoryDependenciesClientBuilder;
+import org.jfrog.build.extractor.docker.DockerAgentUtils;
+import org.jfrog.build.extractor.docker.types.DockerImage;
+import org.jfrog.build.extractor.packageManager.PackageManagerExtractor;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.jfrog.build.extractor.packageManager.PackageManagerUtils.createArtifactoryClientConfiguration;
+
+public class DockerPush extends PackageManagerExtractor {
+    ArtifactoryBuildInfoClientBuilder buildInfoClientBuilder;
+    ArtifactoryDependenciesClientBuilder dependenciesClientBuilder;
+    String imageTag;
+    String deploymentRepository;
+    String host;
+    Log logger;
+    private Map<String, String> env;
+    private Map<String, String> artifactProperties;
+    private String username;
+    private String password;
+    private List<Module> modulesList = new ArrayList<>();
+
+    /**
+     * @param buildInfoClientBuilder    - Build Info client builder.
+     * @param dependenciesClientBuilder - Dependencies client builder.
+     * @param deploymentRepository      - The repository it'll deploy to.
+     * @param imageTag                  - Image tag to push.
+     * @param logger                    - The logger.
+     * @param logger                    - The logger.
+     * @param logger                    - The logger.
+     * @param env                       - Environment variables to use during docker push execution.
+     */
+    public DockerPush(ArtifactoryBuildInfoClientBuilder buildInfoClientBuilder, ArtifactoryDependenciesClientBuilder dependenciesClientBuilder, String imageTag, String host, Map<String, String> artifactProperties, String deploymentRepository, String username, String password, Log logger, Map<String, String> env) {
+        this.buildInfoClientBuilder = buildInfoClientBuilder;
+        this.dependenciesClientBuilder = dependenciesClientBuilder;
+        this.deploymentRepository = deploymentRepository;
+        this.logger = logger;
+        this.imageTag = imageTag;
+        this.host = host;
+        this.username = username;
+        this.password = password;
+        this.env = env;
+        this.artifactProperties = artifactProperties;
+    }
+
+    /**
+     * Allow running docker push using a new Java process.
+     */
+    public static void main(String[] ignored) {
+        try {
+            ArtifactoryClientConfiguration clientConfiguration = createArtifactoryClientConfiguration();
+            // Client builders.
+            ArtifactoryBuildInfoClientBuilder buildInfoClientBuilder = new ArtifactoryBuildInfoClientBuilder().setClientConfiguration(clientConfiguration, clientConfiguration.publisher);
+            ArtifactoryDependenciesClientBuilder dependenciesClientBuilder = new ArtifactoryDependenciesClientBuilder().setClientConfiguration(clientConfiguration, clientConfiguration.publisher);
```

Comment:
ArtifactoryDependenciesClientBuilder should not be used for uploads, only for downloads.

Reply:
We use it to download the images' manifest